packages/core/src/SearchInput.js

```diff
@@ -18,100 +18,148 @@ export const BEM = {
     input: ROOT_BEM.element('input'),
     inputWrapper: ROOT_BEM.element('input-wrapper'),
     resetBtn: ROOT_BEM.element('reset-button'),
+    icon: ROOT_BEM.element('input-icon'),
 };
 
 // a React.Component ensures it can be re-rendered when context changes
 class SearchInput extends Component {
     static propTypes = {
         /**
-         * Use `input` to inject props to the underlying <input>
+         * Use `inputProps` to inject props to the underlying <input>
          */
-        input: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+        inputProps: PropTypes.object, // eslint-disable-line react/forbid-prop-types
         placeholder: PropTypes.string,
         defaultValue: PropTypes.string,
+        value: PropTypes.string,
+        onChange: PropTypes.func,
         onSearch: PropTypes.func,
+        onReset: PropTypes.func,
+        searchWhenInputChange: PropTypes.bool,
```

Comment:
I believe `on` fits here better then `when`
```suggestion
        searchOnInputChange: PropTypes.bool,
```

Reply:
OK.